src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -753,23 +755,10 @@ public void CreateIndex([@NotNull]string name, [@NotNull]IIndex index)
         [Obsolete("CreateIndex is deprecated, please use GetDefaultCollection().CreateIndex")]
         public void CreateIndex([@NotNull] string name, [@NotNull] IndexConfiguration indexConfig)
         {
-            CBDebug.MustNotBeNull(WriteLog.To.Database, Tag, nameof(name), name);
-            CBDebug.MustNotBeNull(WriteLog.To.Database, Tag, nameof(indexConfig), indexConfig);
+            CBDebug.MustNotBeNull(WriteLog.To.Database, Tag, nameof(DefaultCollection), DefaultCollection);
 
-            ThreadSafety.DoLocked(() =>
-            {
-                CheckOpen();
-                LiteCoreBridge.Check(err =>
-                {
-                    var internalOpts = indexConfig.Options;
-                    // For some reason a "using" statement here causes a compiler error
-                    try {
-                        return Native.c4db_createIndex2(c4db, name, indexConfig.ToN1QL(), indexConfig.QueryLanguage, indexConfig.IndexType, &internalOpts, err);
-                    } finally  {
-                        internalOpts.Dispose();
-                    }
-                });
-            });
+            DefaultCollection.CheckCollectionValid();
```

Comment:
This stuff all happens again anyway inside of CreateIndex, etc right?  If so, this is redundant and sort of an antipattern.  There should be no need to call CheckCollectionValid outside of the collection class.

Reply:
Removed the checks from Database